crates/prompt_gateway/src/stream_context.rs

```diff
@@ -892,9 +893,37 @@ impl StreamContext {
         let endpoint = prompt_target.endpoint.unwrap();
         let path: String = endpoint.path.unwrap_or(String::from("/"));
 
+        // only add params that are of string, number and bool type
```

Comment:
this makes sense to me. We should add this in the documentation as well - so that its clear to developers what we swap into the parameters. Btw, how do we distinguish between string/number/bool - aren't they all represented as strings in the path?

Reply:
Other options are list (or array) and dict (objects). That wouldn't make sense for path param

Reply:
> how do we distinguish between string/number/bool

In yaml and in json string, number and bool are stored differently. For example,

```yaml
format: huggingface
```

here huggingface is a string.

Also this is same,

```yaml
format: "huggingface"
```

In following example 100 is a number and true is bool

```yaml
tracing:
  random_sampling: 100
  trace_arch_internal: true
```